### PR TITLE
Update resource spec to 2.22.0

### PR DESCRIPTION
**Pull request by cmmeyer**
*Issue #, if available:*

*Description of changes:*
Spec URL: https://s3.us-east-1.amazonaws.com/cfn-resource-specifications-us-east-1-prod/2.22.0/CloudFormationResourceSpecification.json
2.22.0, 2019-02-08, 351
  New:
* AWS::ApiGatewayV2::Api
* AWS::ApiGatewayV2::Authorizer
* AWS::ApiGatewayV2::Deployment
* AWS::ApiGatewayV2::Integration
* AWS::ApiGatewayV2::IntegrationResponse
* AWS::ApiGatewayV2::Model
* AWS::ApiGatewayV2::Route
* AWS::ApiGatewayV2::RouteResponse
* AWS::ApiGatewayV2::Stage

**Note:** This update required additional patching due to malformed and undocumented resource properties:

* AWS::EC2::LaunchTemplate.CapacityReservationPreference
* AWS::CodeBuild::Project.FilterGroup

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/data/CloudSpecs/ap-northeast-1.json [hunk 1]
```diff
@@ -6213,6 +6287,54 @@
         }
       }
     },
+    "AWS::EC2::LaunchTemplate.CapacityReservationPreference": {
```
Comment (kddejong) [inline src/cfnlint/data/CloudSpecs/ap-northeast-1.json]: This seems wrong.

Comment (cmmeyer) [reply src/cfnlint/data/CloudSpecs/ap-northeast-1.json]: Yeah -- I'm having a heck of a time untangling this. looks like there were several undocumented property types injected into the spec around capacity.  My Saturday morning brain is having challenges patching around them.

Comment (kddejong) [reply src/cfnlint/data/CloudSpecs/ap-northeast-1.json]: Yea no rush. I can help decouple it on Monday. It doesn’t look easy to decouple.

Comment (cmmeyer) [reply src/cfnlint/data/CloudSpecs/ap-northeast-1.json]: Patched it (and also AWS::CodeBuild::Project.FilterGroup)

**Pull request merged**